packages/core/src/routes/swagger.ts

```diff
@@ -45,7 +50,40 @@ export default function swaggerRoutes<T extends AnonymousRouter>(router: T) {
             ])
           ),
         ];
-      })
+      }
+    );
+
+    /**
+     * Avoid `Object.fromEntries()` overwriting the former `path1` entry with the latter one:
+     * e.g. convert
+     * [
+     *   [ path1, { head: { ... }, get: { ... } } ],
+     *   [ path2, { post: { ... }, patch: { ... } } ],
+     *   [ path1, { delete: { ... } } ],
+     *   ...
+     * ]
+     * to
+     * {
+     *   path1: { head: { ... }, get: { ... }, delete: { ... } } ,
+     *   path2: { post: { ... }, patch: { ... } } ,
+     *   ...
+     * }
+     */
```

Comment:
This doesn't feel right.  As the original route schema flatten the methods internally, why do we still have the cases you listed above? Should be handled at the root route level